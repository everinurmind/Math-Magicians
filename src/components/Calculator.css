.calculator {
  width: 440px;
  height: 360px;
  border: 1px solid #ccc;
  display: flex;
  flex-direction: column;
}

.display {
  height: 60px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding: 0 5px;
  font-size: 20px;
  background-color: #8e8e8e;
  color: #fff;
}

.keypad {
  display: flex;
  flex-direction: column;
}

.row {
  display: flex;
}

.key {
  height: 60px;
  width: 110px;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 16px;
  background-color: #ccc;
  border: none;
  cursor: pointer;
  transition: background-color 0.2s ease;
  box-shadow: 0 0 0 0.5px #a1a1a1;
}

.key:hover {
  background-color: #ddd;
}

.operator {
  background-color: #ff8c00;
}

.operator:hover {
  background-color: #ff6300;
}

.zero {
  width: 50%;
}
