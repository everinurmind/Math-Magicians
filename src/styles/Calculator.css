/* Calculator Styling */
.container {
  display: flex;
  flex-direction: row;
  gap: 200px;
}

.header {
  font-size: 40px;
  font-weight: 900;
  margin-top: -40px;
}

.calculator {
  width: 440px;
  height: 360px;
  margin-left: 100px;
  border: 1px solid #ccc;
  display: flex;
  flex-direction: column;
  border-radius: 20px;
}

.display {
  height: 60px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding: 0 15px;
  font-size: 20px;
  background-color: #8e8e8e;
  color: #fff;
  border-top-right-radius: 20px;
  border-top-left-radius: 20px;
}

.keypad {
  display: flex;
  flex-direction: column;
}

.row {
  display: flex;
}

.key {
  height: 60px;
  width: 110px;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 16px;
  background-color: #ccc;
  border: none;
  cursor: pointer;
  transition: background-color 0.2s ease;
  box-shadow: 0 0 0 0.5px #a1a1a1;
}

.key:hover {
  background-color: #ddd;
}

.operator {
  background-color: #ff8c00;
}

.operator2 {
  border-bottom-right-radius: 20px;
  background-color: #ff8c00;
}

.operator:hover {
  background-color: #ff6300;
}

.operator2:hover {
  background-color: #ff6300;
}

.zero {
  width: 50%;
  border-bottom-left-radius: 20px;
}
