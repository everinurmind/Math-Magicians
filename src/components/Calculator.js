import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';
import Display from './Display';

function Calculator() {
  const [calculatorState, setCalculatorState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    const newState = calculate(calculatorState, buttonName);
    setCalculatorState(newState);
  };

  return (
    <div className="calculator">
      <Display value={calculatorState.next || calculatorState.total || '0'} />
      <Display value={calculatorState.operation || ''} />
      <div className="keypad">
        <div className="row">
          <button type="button" className="key" onClick={() => handleClick('AC')}>AC</button>
          <button type="button" className="key" onClick={() => handleClick('+/-')}>+/-</button>
          <button type="button" className="key" onClick={() => handleClick('%')}>%</button>
          <button type="button" className="key operator" onClick={() => handleClick('รท')}>รท</button>
        </div>
        <div className="row">
          <button type="button" className="key" onClick={() => handleClick('7')}>7</button>
          <button type="button" className="key" onClick={() => handleClick('8')}>8</button>
          <button type="button" className="key" onClick={() => handleClick('9')}>9</button>
          <button type="button" className="key operator" onClick={() => handleClick('x')}>x</button>
        </div>
        <div className="row">
          <button type="button" className="key" onClick={() => handleClick('4')}>4</button>
          <button type="button" className="key" onClick={() => handleClick('5')}>5</button>
          <button type="button" className="key" onClick={() => handleClick('6')}>6</button>
          <button type="button" className="key operator" onClick={() => handleClick('-')}>-</button>
        </div>
        <div className="row">
          <button type="button" className="key" onClick={() => handleClick('1')}>1</button>
          <button type="button" className="key" onClick={() => handleClick('2')}>2</button>
          <button type="button" className="key" onClick={() => handleClick('3')}>3</button>
          <button type="button" className="key operator" onClick={() => handleClick('+')}>+</button>
        </div>
        <div className="row">
          <button type="button" className="key zero" onClick={() => handleClick('0')}>0</button>
          <button type="button" className="key" onClick={() => handleClick('.')}>.</button>
          <button type="button" className="key operator" onClick={() => handleClick('=')}>=</button>
        </div>
      </div>
    </div>
  );
}

export default Calculator;
