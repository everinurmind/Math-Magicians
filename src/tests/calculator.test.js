import renderer from 'react-test-renderer';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Calculator from '../components/Calculator';

test('should render Calculator.js', () => {
  const text = renderer.create(
    <BrowserRouter>
      <Routes>
        <Route path="/calculator" element={<Calculator />} />
      </Routes>
    </BrowserRouter>,
  );

  expect(text.toJSON()).toMatchSnapshot();
});
