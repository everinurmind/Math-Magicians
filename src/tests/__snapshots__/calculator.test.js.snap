// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Calculator.js 1`] = `null`;
